crates/rust-analyzer/src/config.rs

```diff
@@ -903,43 +802,48 @@ impl Config {
                 serde_json::Error::custom("expected a non-empty string"),
             ));
         }
-        config
+        (config, should_update)
+    }
+
+    pub fn apply_change_whatever(&self, change: ConfigChange) -> (Config, ConfigError) {
```

Comment:
The whatever was me being too lazy to come up with a proper name here 😅 Let's call this `apply_change` and rename the other to `apply_change_with_sink` and migrate the callers of that function to this one (probably need to change the return type to `(Config, ConfigError, bool)` then. I don' think any caller cares about setting up the error sink themselves (so we could probably get rid of the public API that takes the error sink as well)